Project/src/main/java/app/FileUploadController.java

```diff
@@ -38,4 +51,23 @@ public String uploadFile(@RequestParam("file") MultipartFile file) throws IOExce
         return "redirect:/upload";
     }
 
+    /***
+     * Endpoint to download a file from the database
+     * @param id The ID of the file that is to be downloaded
+     * @return
+     */
+    @GetMapping("/getfile/{id}")
+    public ResponseEntity<byte[]> getFileById(@PathVariable(value = "id")Long id){
+        Optional<File> fileOptional = repository.findById(id);
```

Comment:
Should we add a check to ensure that the user trying to download the file is authorized to do so?